src/index.js

```diff
@@ -30,6 +30,36 @@ class RpcEngine extends SafeEventEmitter {
   // Private
   //
 
+  async _handleBatch (reqs, cb) {
+    const batchRes = []
+    for (const r of reqs) {
+      try {
+        let [err, res] = await this._promiseHandle(r)
+        if (!res) {
+          if (err) {
+            throw err
```

Comment:
Sometimes `res.error` is populated. Should we throw on that too?

Reply:
No, individual responses with `error` should be added to the batch request array in the same way as responses with `result`. That condition is there for defensive programming purposes, in case something goes seriously wrong.

In my reading, unless it's a fatal `json-rpc-engine` internal error, there should always be an error on the response due to `_runMiddlewareDown`. If not, this would be implemented correctly.